text/0119-dynamic-component-creation.md

```diff
@@ -0,0 +1,335 @@
+
+- **The constructor is `undefined` or `null`**
+    - When the constructor is undefined the `<lwc.dynamic>` tag along with all of its children are not rendered to the DOM.
+- **The constructor is defined and is not a `LightningElementConstructor`**
+    - When the value provided to `lwc:is` is not a `LightningElementConstructor` then an error is thrown indicating the constructor value must be `LightningElementConstructor`.
+- **The constructor is defined and is a `LightningElementConstructor`**
+    - When the constructor is a `LightningElementConstructor`, the component will be rendered in place of `<lwc.dynamic>`. The tag name used for the dynamic component is the value of `lightningElementName` on `LightningElementConstructor`.
+
+_Note in the case of lazy loading, the component author is responsible for resolving the promise.  The only value that can be given to `lwc:is` is a `LightningElementConstructor`._
+
+#### Selecting the dynamic component
+
+It is important to note that `<lwc.dynamic>` will not be rendered to the DOM.  As such, it is not available to be selected using `this.template.querySelector`.  The actual custom element must be selected once it has been rendered to the DOM.  The dynamic component can be selected either through the `lightningElementName` property on the constructor or through another attribute assigned to the component.
+
+Component authors can use the `connectedCallback` of the dynamic component to signal when the component has been rendered to the DOM.
+
+#### Assigning attributes and event listeners
+
+Attributes and event listeners can be passed to the `<lwc.dynamic>` tag declaratively or can be assigned directly to the custom element imperatively.
+
+Attributes and event listeners assigned declaratively will be assigned to the dynamic component once it has been created.
+
+Declaratively:
+
+```html
+<template>
+    <lwc.dynamic lwc:is={ctor} class="container" onclick={handleClick}></lwc.dynamic>
+</template>
+```
+
+Imperatively:
+
+```javascript
+import { LightningElement } from "lwc";
+
+export default class DynamicCtor extends LightningElement {
+  customCtor;
+
+  connectedCallback() {
+    this.loadCtor();
+  }
+
+  async loadCtor() {
+    const { default: ctor } = await import("c/customConstructor");
+    this.customCtor = ctor;
+    this.template.querySelector(ctor.lightningElementName).setAttribute('title', 'hawaii');
+    this.template.querySelector(ctor.lightningElementName).addEventListener('click', () => console.log('hello world'));
+  }
+}
+```
+#### Usage with other LWC directives
+
+All [LWC custom element directives](https://lwc.dev/guide/reference#directives-for-nested-templates) with the exception of `lwc:external` are available to use on `<lwc.dynamic>` and their functionality will be passed through to the custom element once it has been created.
+
+`lwc:external` will not be available because the constructor provided to the `lwc:is` directive must be a `LightningElementConstructor`.
```

Comment:
We should add support for `lwc:ref`, as mentioned above. 🙂

Reply:
it will definitely work with this proposal... make sure we have an example.

Reply:
Added an [example](https://github.com/salesforce/lwc-rfcs/pull/71/files#diff-a1de05148a61ea9b00f70bd4ff5feb3895c16dd99707b173ba57c081b859250bR233-R258) using `lwc:ref`